magnus_timesheet/models/analytic.py

```diff
@@ -432,12 +435,12 @@ def write(self, vals):
                         'Please fill in Fee Rate Product in employee %s.\n '
                     ) % user.name)
                 vals['product_id'] = product_id
-            ts_line = vals.get('ts_line', self.ts_line)
+            ts_line = vals.get('ts_line', self.product_uom_id == uom_hour)
```

Comment:
This (and below) is the effective change, I kept the others because higher versions of Odoo complain when you have inconsistent stored/nonstored fields. The slices are necessary because on the update I had a case where the function returns two records.

What's going on here is that core does a write during [create]( https://github.com/OCA/OCB/blob/10.0/addons/sale/models/sale_analytic.py#L141) of analytic lines (this could be more efficient by checking if res is different from vals, but that's another issue), at this point computed fields aren't yet computed. So accessing the field in our code will recompute it, and as we're in create, this happens per record.  And then another time in bulk when the create of the sheet is done.